### PR TITLE
Implement GetDiskFreeSpaceExA() / GetDiskFreeSpaceA()

### PR DESCRIPTION
**Pull request by thrimbor**
This implements `GetDiskFreeSpaceExA()` and `GetDiskFreeSpaceA()`.
I checked the results of both functions against the values reported by the dashboard on my Xbox, and they do match. The error reporting (both `SetLastError(ERROR_PATH_NOT_FOUND);` and returning `0xFFFFFFFF` in `GetDiskFreeSpaceA()`) were checked against the behavior of Windows 7.

### CODE REVIEW

File: lib/winapi/filemanip.c [hunk 1]
```diff
@@ -235,3 +236,106 @@ BOOL MoveFileA (LPCTSTR lpExistingFileName, LPCTSTR lpNewFileName)
...
+        *lpSectorsPerCluster = fsSizeInfo.SectorsPerAllocationUnit;
+    }
+    if (lpBytesPerSector) {
+        *lpBytesPerSector = fsSizeInfo.BytesPerSector;
+    }
+    if (lpNumberOfFreeClusters) {
+        *lpNumberOfFreeClusters = fsSizeInfo.AvailableAllocationUnits.HighPart == 0 ? fsSizeInfo.AvailableAllocationUnits.LowPart : 0xFFFFFFFF;
+    }
+    if (lpTotalNumberOfClusters) {
+        *lpTotalNumberOfClusters = fsSizeInfo.TotalAllocationUnits.HighPart == 0 ? fsSizeInfo.TotalAllocationUnits.LowPart : 0xFFFFFFFF;
```
Comment (JayFoxRox) [inline lib/winapi/filemanip.c]: nit: `fsSizeInfo.TotalAllocationUnits.HighPart ? 0xFFFFFFFF : fsSizeInfo.TotalAllocationUnits.LowPart;`

This helps with readability in my opinion (as the connection between highpart and maxing out the value is more apparent). It's also shorter and doesn't add more operations into the ternary operator (which users often struggle with anyway).
Also matches more closely with the implicit `!= NULL` in the assert for the pathname above.

*(Other instance, too)*

**Comment by JayFoxRox** [comment]
The code looks good (as usual). Waiting for people to test this (or add their own reviews).

File: lib/winapi/filemanip.c (hunk 1 above)
Comment (thrimbor) [reply lib/winapi/filemanip.c]: Good point, I changed it.

**Comment by dracc** [comment]
Tested, both `GetDiskFreeSpace` and `GetDiskFreeSpaceEx` confirmed working on my 1.6.
Returns the expected amount of free storage for the controlled folder. (`F:\Games\`)

Code also looks good. :+1:

File: lib/winapi/filemanip.c [hunk 2]
```diff
@@ -235,3 +236,106 @@ BOOL MoveFileA (LPCTSTR lpExistingFileName, LPCTSTR lpNewFileName)
...
+BOOL GetDiskFreeSpaceA (LPCSTR lpRootPathName, LPDWORD lpSectorsPerCluster, LPDWORD lpBytesPerSector, LPDWORD lpNumberOfFreeClusters, LPDWORD lpTotalNumberOfClusters)
+{
+    NTSTATUS status;
+    HANDLE handle;
+    ANSI_STRING path;
+    OBJECT_ATTRIBUTES objectAttributes;
+    IO_STATUS_BLOCK ioStatusBlock;
+    FILE_FS_SIZE_INFORMATION fsSizeInfo;
+
+    assert(lpRootPathName);
```
Comment (dracc) [inline lib/winapi/filemanip.c]: "If this parameter is NULL, the function uses the root of the current disk."
https://docs.microsoft.com/en-us/windows/win32/api/fileapi/nf-fileapi-getdiskfreespacea

Same applies to `GetDiskFreeSpaceEx`.

Comment (thrimbor) [reply lib/winapi/filemanip.c]: True, but we don't support the concept of current disks / current directories yet.

Comment (dracc) [reply lib/winapi/filemanip.c]: Makes sense.
I personally wouldn't want the MS docs behavior anyways tbh.

Comment (JayFoxRox) [reply lib/winapi/filemanip.c]: We probably do want to support it in the future, so we can set the current directory to `D:\` and then we can probably use relative paths again?

If we don't follow MS docs we'll usually document it. However, in this case we can make an exception (as there's no way to set the current folder so it's obviously problematic). Also the `assert` explicitly handles it.
I guess ideally we'd add the sentence from the MSDN to that `assert` so if people trigger it in a naive port they'll immediately know what's going on.

*(I had actually noticed this in my initial review, too but was fine with this)*

**Comment by dracc** [comment]
Looks good to me, test results turned out as expected. :muscle:

**Pull request merged**